superset/sql_parse.py

```diff
@@ -319,12 +320,17 @@ def _extract_tables_from_statement(self, statement: exp.Expression) -> set[Table
         elif isinstance(statement, exp.Command):
             # Commands, like `SHOW COLUMNS FROM foo`, have to be converted into a
             # `SELECT` statetement in order to extract tables.
-            literal = statement.find(exp.Literal)
-            if not literal:
+            if not (literal := statement.find(exp.Literal)):
                 return set()
 
-            pseudo_query = parse_one(f"SELECT {literal.this}", dialect=self._dialect)
-            sources = pseudo_query.find_all(exp.Table)
+            try:
```

Comment:
This issue surfaced when trying to parse an `ADD JAR ...` statement. which `sqlglot` believes contains literal expressions. The TL;DR is `parse_one` can throw and we should catch these errors (similar to when using the `sqlglot.parse` method).

Reply:
i have no idea what your use case is, but if you want more lenient sql parsing, you can try error_level=IGNORE

Reply:
Ha! I was wondering whether I should have raised this with you.

Reply:
always free to chat. you can hit me up on slack (i'm in your slack or you can come to mine)

Reply:
Aside, but I had a similar problem with the old `sqlparse` solution with giving CTEs names which are keywords in other engines; I'm curious if swapping to sqlglot has fixed that actually, since it's more engine-aware. Specifically I made the mistake of using the name `ref` for the CTE in a trino query, and `REF` is apparently a keyword in some engines (not trino as far as I'm aware). I'll give it a go with the new sqlglot approach sometime.